mir-ci/mir_ci/test_drag_and_drop.py

```diff
@@ -40,27 +53,43 @@ class TestDragAndDrop:
         ],
     )
     async def test_source_and_dest_match(self, modern_server, app) -> None:
-        modern_server = DisplayServer(modern_server, add_extensions=VirtualPointer.required_extensions)
-        program = modern_server.program(apps.App(app))
+        extensions = VirtualPointer.required_extensions + ScreencopyTracker.required_extensions
+        server_instance = DisplayServer(modern_server, add_extensions=extensions)
+        program = server_instance.program(apps.App(app))
+
+        robot_settings = dedent(
+            f"""\
+            Library    {MIR_CI_PATH}/robot_libraries/WaylandHid.py
+            Library    {MIR_CI_PATH}/robot_libraries/Screencopy.py
+            Resource   {MIR_CI_PATH}/robot_resources/screencopy.resource
+        """
+        ).strip("\n")
 
-        robot_test_case = f"""Source and Destination Match
-            Sleep     {STARTUP_TIME}
-            Move Pointer To Absolute    40    40
-            Sleep     {A_SHORT_TIME}
-            Press LEFT Button
-            Sleep     {A_SHORT_TIME}
-            Move Pointer To Absolute    120    70
-            Sleep     {A_SHORT_TIME}
-            Move Pointer To Absolute    200    100
-            Sleep     {A_SHORT_TIME}
-            Release LEFT Button
+        robot_test_case = dedent(
+            f"""\
+            Source and Destination Match
+                Sleep     {STARTUP_TIME}
+                ${{regions}} =    Test Match    {MIR_CI_PATH}/robot_templates/drag_and_drop_src.png
+                ${{center}} =    Get Center    ${{regions}}[0]
+                Move Pointer To Absolute    ${{center}}[x]    ${{center}}[y]
+                Sleep     {A_SHORT_TIME}
+                Press LEFT Button
+                ${{off_center}} =    Add Displacement    ${{center}}    20    20
+                Move Pointer To Absolute    ${{off_center}}[x]    ${{off_center}}[y]
+                Sleep     {A_SHORT_TIME}
+                ${{regions}} =    Test Match    {MIR_CI_PATH}/robot_templates/drag_and_drop_dst.png
+                ${{center}} =    Get Center    ${{regions}}[0]
+                Move Pointer To Absolute    ${{center}}[x]    ${{center}}[y]
+                Sleep     {A_SHORT_TIME}
+                Release LEFT Button
         """
+        ).strip("\n")
 
         with tempfile.NamedTemporaryFile(mode="w+", suffix=".robot", buffering=1) as robot_file:
-            robot_file.write(ROBOT_TEMPLATE.format(library_path=ROBOT_LIBRARY_PATH, test_case=robot_test_case))
-            robot = modern_server.program(apps.App(("robot", "-o", "NONE", "-r", "NONE", robot_file.name)))
+            robot_file.write(ROBOT_TEMPLATE.format(settings=robot_settings, test_case=robot_test_case))
+            robot = server_instance.program(apps.App(("robot", "-d", f"{MIR_CI_PATH}/robot_log/", robot_file.name)))
```

Comment:
We should avoid writing anything in `MIR_CI_PATH`, as it could affect other tests. Wonder if we should record it as a property of the test so it ends up in the XML.

Reply:
True. Although I don't know how to record these logs as a property of the test, I know how to create a subdir named after each node and write the log files there.

Reply:
Right, and after you do this, back in pytest you can use the `record_property` fixture to feed that into the XML result:

https://docs.pytest.org/en/4.6.x/reference.html#record-property

There's some uses of this across Benchmarker, DisplayServer and test_screencopy_bandwidth.

Reply:
Ah but that can be left for later. For now just push it to a temporary folder.

Actually just realized there're fixtures for that:

https://docs.pytest.org/en/latest/how-to/tmp_path.html

Reply:
Cool, I'll look into it.